routes/login/login.controller.js

```diff
@@ -1,27 +1,63 @@
-module.exports = function(app) {
+module.exports = function (app) {
   // redirect from "/login" → "/login/accessCode"
   app.get('/login', (req, res) => res.redirect('/login/code'))
   app.get('/login/code', (req, res) =>
     res.render('login/code', { title: 'Enter access code', data: req.session || {} }),
   )
-  app.post('/login/code', postCode)
+  app.post('/login/code', processCode)
   app.get('/login/success', (req, res) =>
     res.render('login/success', { title: 'Your access code', data: req.session || {} }),
   )
+
+  //SIN
+  app.get('/login/sin', (req, res) =>
+    res.render('login/sin', { title: 'Enter your SIN', data: req.session || {} }),
+  )
+  app.post('/login/sin', processSIN)
+
+  app.get('/login/sin', (req, res) =>
+    res.render('login/sin', { title: 'Enter your SIN', data: req.session || {} }),
+  )
+
 }
 
-const postCode = (req, res) => {
+//POST functions that handle setting the login data in the session and handle redirecting to the next page or sending an error to the client.
+//Note that this is not the only error validation, see routes defined above.
+
+const validateRedirect = (req, res) => {
   let redirect = req.body.redirect || null
   if (!redirect) {
     throw new Error(`[POST ${req.path}] 'redirect' parameter missing`)
   }
+  return redirect
+}
```

Comment:
Yeah, I was thinking of adding this in as middleware so it's more generalized.

Check it out here: https://expressjs.com/en/guide/writing-middleware.html

And I have an example implementation checking that someone's logged in from the cra-alpha app: https://github.com/cds-snc/cra-alpha/search?q=checklogin&unscoped_q=checklogin

Reply:
I started looking into this, but I think I'll split it up in another PR since it changes quite a few things that aren't for this page.

Reply:
Okay, sounds good.